python/lsst/jointcal/jointcal.py

```diff
@@ -216,22 +230,23 @@ def run(self, ref, tract):
         chi2 = fit.ComputeChi2()
         print(chi2)
 
-        for i in range(20) :
-            r = fit.Minimize("Distortions Positions",5) # outliers removal at 5 sigma.
+        for i in range(20):
+            r = fit.Minimize("Distortions Positions", 5)  # outliers removal at 5 sigma.
```

Comment:
Le 21/06/2016 20:52, John Parejko a écrit :
>
> In python/lsst/jointcal/jointcal.py 
> <https://github.com/lsst/jointcal/pull/12#discussion_r67929605>:
>
> > @@ -216,22 +230,23 @@ def run(self, ref, tract):
> >          chi2 = fit.ComputeChi2()
> >          print(chi2)
> >
> > -        for i in range(20) :
> > -            r = fit.Minimize("Distortions Positions",5) # outliers removal at 5 sigma.
> > +        for i in range(20):
> > +            r = fit.Minimize("Distortions Positions", 5)  # outliers removal at 5 sigma.
>
> I think it's saying "minimize both of these quantities", but I'm not 
> going to touch it for now.
>
Yes, it exactly means to minimize both sets of parameters simultaneously.
I've never tried to do that right away. You'd better have before something
like

r = fit.Minimize("Distortions")
r = fit.Minimize("Positions")
  

I have opted for using a string for controlling the fit, because
depending on the model you are fitting, you might have
subset of parameters that you could want to control independently,
in particular in the initialization the phase. A string is flexible 
enough to do that.

by the way, if you prefer writing "Distortions+Positions" for sake of 
clarity,
it will just do the same thing because parameter sets are freed based
on a logic "à la grep". And if you have a better name for "Distortions", 
please
suggest.

> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/lsst/jointcal/pull/12/files/65ca60520b228b7dd6c48f7a6604b94e7469e89f#r67929605>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AMQrtrFr1oUcPAtxFkDkrD5hzbiRboEwks5qODMBgaJpZM4I6Kum>.
>


-- 


                   -------------------
Pierre Astier , LPNHE, 12-22 1er étage.
4 place Jussieu, F 75252 Paris Cedex 05
tel (33) 1 44 27 76 47  ---- fax (33) 1 44 27 46 38